sycl/CMakeLists.txt

```diff
@@ -90,6 +90,12 @@ if( NOT OpenCL_LIBRARIES )
         "${LLVM_LIBRARY_OUTPUT_INTDIR}/${CMAKE_SHARED_LIBRARY_PREFIX}OpenCL${CMAKE_SHARED_LIBRARY_SUFFIX}")
     set(OPENCL_ICD_LOADER_WDK "")
   endif()
+  if (CMAKE_C_COMPILER)
+    set(OPENCL_C_COMPILER "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}")
```

Comment:
Using 'OPENCL_C_COMPILER' variable name is a bad idea, unless you specifically want confuse people.

I would refactor the code into something like this:
```cmake
if (MSVC)
  set(OpenCL_LIBRARIES ...)
  set(AUX_CMAKE_ARGS " -DOPENCL_ICD_LOADER_REQUIRE_WDK=OFF")
else ()
  set(OpenCL_LIBRARIES ...)
  // NOTE:   set(AUX_CMAKE_ARGS "") is not needed. set(OPENCL_ICD_LOADER_WDK "") is useless already.
endif ()

if (CMAKE_C_COMPILER)
  set(AUX_CMAKE_ARGS " -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}")
endif()
if (CMAKE_CXX_COMPILER)
  set(AUX_CMAKE_ARGS " -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}")
endif()

ExternalProject_Add(ocl-icd
...
CMAKE_ARGS ...
   ${AUX_CMAKE_ARGS}
)
```